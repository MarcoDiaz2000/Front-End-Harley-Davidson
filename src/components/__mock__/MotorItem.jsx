import React from 'react';

const MotorItem = () => (
  <div className="w-full h-96 card">
    <div className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2">
      <img className="w-80 h-auto" src="src/img" alt="motor" />
    </div>
    <h1 className="font-bold text-white bg-blur">Model 1</h1>
    <a
      href="/motors/1"
      className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
    >
      See Model
    </a>
  </div>
);
export default MotorItem;
