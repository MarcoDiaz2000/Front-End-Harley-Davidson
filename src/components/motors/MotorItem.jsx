import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const MotorItem = ({ motor }) => (
  <div className="w-full h-96 card">
    <div className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2">
      <img className=" max-h-full" src={motor.image_url} alt="motor" />
    </div>
    <h1 className="font-bold text-white bg-blur">{motor.name}</h1>
    <NavLink
      to={`/motors/${motor.id}`}
      type="button"
      className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
    >
      See Model
    </NavLink>
  </div>
);
export default MotorItem;

MotorItem.propTypes = {
  motor: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    image_url: PropTypes.string.isRequired,
  }).isRequired,
};
