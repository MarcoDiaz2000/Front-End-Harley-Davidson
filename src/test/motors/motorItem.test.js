import React from 'react';
import renderer from 'react-test-renderer';
import MotorItem from '../../components/__mock__/MotorItem';

describe('MotorItem', () => {
  it('renders correctly', () => {
    const component = renderer.create(<MotorItem />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
