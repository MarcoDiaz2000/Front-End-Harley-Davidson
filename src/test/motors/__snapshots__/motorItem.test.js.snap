// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MotorItem renders correctly 1`] = `
<div
  className="w-full h-96 card"
>
  <div
    className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2"
  >
    <img
      alt="motor"
      className="w-80 h-auto"
      src="src/img"
    />
  </div>
  <h1
    className="font-bold text-white bg-blur"
  >
    Model 1
  </h1>
  <a
    className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
    href="/motors/1"
  >
    See Model
  </a>
</div>
`;
